src/gui/widgets/LcdFloatSpinBox.cpp

```diff
@@ -0,0 +1,254 @@
+/*
+ * LcdFloatSpinBox.cpp - class LcdFloatSpinBox (LcdSpinBox for floats)
+ *
+ * Copyright (c) 2005-2014 Tobias Doerffel <tobydox/at/users.sourceforge.net>
+ * Copyright (c) 2008 Paul Giblock <pgllama/at/gmail.com>
+ * Copyright (c) 2020 Martin Pavelek <he29.HS/at/gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "LcdFloatSpinBox.h"
+
+#include <cmath>
+#include <QApplication>
+#include <QHBoxLayout>
+#include <QInputDialog>
+#include <QLabel>
+#include <QMouseEvent>
+#include <QPainter>
+#include <QPixmap>
+#include <QStyleOptionFrameV2>
+#include <QVBoxLayout>
+
+#include "CaptionMenu.h"
+#include "embed.h"
+#include "GuiApplication.h"
+#include "gui_templates.h"
+#include "MainWindow.h"
+
+
+LcdFloatSpinBox::LcdFloatSpinBox(int numWhole, int numFrac, QWidget* parent, const QString& name) :
+	FloatModelView(new FloatModel(0, 0, 0, 0, nullptr, name, true), this),
+	m_wholeDisplay(numWhole, parent, name, false),
+	m_fractionDisplay(numFrac, parent, name, true),
+	m_mouseMoving(false),
+	m_intStep(false),
+	m_origMousePos(),
+	m_displayOffset(0)
+{
+	layoutSetup();
+}
```

Comment:
Again running into problems with the `style` parameter: if I delegate it to the other constructor, I have to pass something to the `style`, meaning that the default value defined in `LcdWidget` won't be used. I could copy it here, but then it would be in two different places that have to be kept in mind when the default changes.